### ISSUE TITLE
No way to stop continuous running once its started

### ISSUE THREAD

**Issue by RichardLuo0** [matepek/vscode-catch2-test-adapter#398]
**Checklist**

- [X] The issue is about this extension and NOT about a fork.
- [X] Checked the [ALL the SUPPORT](https://github.com/matepek/vscode-catch2-test-adapter/blob/master/documents/support.md) document.
- [X] The latest version of the extension was used.
- [X] It is not related to remote-vscode or I checked the following [issue](https://github.com/matepek/vscode-catch2-test-adapter/issues/201)
- [X] Imagine yourself into my position and think how hard to debug the issue without insufficient information.
      I understand that you have privacy concerns and I expect you to understand that this extension is developed for free.
      Thanks.

**Describe the bug**
Once the continuous running is triggered when the executable changes, there appears to be a `stop` button. But clicking it doesn't stop anything (the process is still running). The only workaround I found is to turn off continuous running and then turn it back on.
![image](https://github.com/matepek/vscode-catch2-test-adapter/assets/22096448/bc145d95-e407-4a7a-8082-6d2f0d7a5acf)
![image](https://github.com/matepek/vscode-catch2-test-adapter/assets/22096448/6605c638-df9c-440d-90a1-f8eec13611f8)

Also, there is no such button in this panel.
![image](https://github.com/matepek/vscode-catch2-test-adapter/assets/22096448/cec6a89e-680e-4261-922d-ac42b9843467)

However, if I start it manually, everything works fine.

PS. Btw, how do I enable continuous running by default at startup? There seems to be no option.

**To Reproduce**
1. write a program with an infinite loop
2. write a cmakelist.
3. enable `continuous running`
4. compile it and see then try to stop it with the `stop` button.

**Desktop**

- Extension Version: 4.6.0
- VS Code Version: 1.83.1
- Catch2 / Google Test / DOCTest Version: mingw-w64-clang-x86_64-gtest 1.14.0-1 from msys2
- OS Type and Version: Windows 11
- Using remote-ssh/docker/wsl?: no

**Regression bug?**

- Last extension version in which the feature were working:

<details>
  <summary>**Log** (optional but recommended)</summary>

```js
[2023-10-15 07:01:46.302] [INFO] Test naive.0 has started.
[2023-10-15 07:01:46.353] [INFO] watcher event: d:\Repository\leetCode\build-win32-Debug\PSubstrings.exe.TestMate.testListCache.xml
[2023-10-15 07:01:47.354] [INFO] possibly new suite: d:\Repository\leetCode\build-win32-Debug\PSubstrings.exe.TestMate.testListCache.xml
[2023-10-15 07:01:48.255] [DEBUG] setPriority is done priority(16) pid(15112)
[2023-10-15 07:01:48.327] [INFO] Not found running build related process
[2023-10-15 07:02:03.566] [INFO] proc close: d:\Repository\leetCode\build-win32-Debug\PSubstrings.exe [ 1, null ]
[2023-10-15 07:02:03.567] [ERROR] builder should not left behind if no problem <ref *1> f {
  shared: {
    shared: {
      workspaceFolder: { uri: [s], name: 'leetCode', index: 0 },
      log: { _logger: [Object] },
      testController: { controller: [Object], testItem2test: [WeakMap] },
      executeTask: [Function (anonymous)],
      sendRetireEvent: [Function: d],
      varToValue: [
        [Object], [Object],
        [Object], [Object],
        [Object], [Object],
        [Object], [Object]
      ],
      rngSeed: null,
      execWatchTimeout: 10000,
      _execRunningTimeout: 600000,
      execParsingTimeout: 5000,
      isNoThrow: false,
      enabledTestListCaching: true,
      enabledStrictPattern: null,
      googleTestTreatGMockWarningAs: 'nothing',
      googleTestGMockVerbose: 'default',
      hideUninterestingOutput: true,
      _execRunningTimeoutChangeEmitter: c {
        x: 1,
        f: undefined,
        g: undefined,
        j: undefined,
        u: undefined,
        q: [Function (anonymous)],
        t: [s]
      },
      _cancellationTokenSource: L { f: [R], g: undefined },
      cancellationToken: R { a: false, b: null },
      onDidChangeExecRunningTimeout: [Function (anonymous)],
      taskPool: { _maxTaskCount: 1, _runningTaskCount: 1, _waitingTasks: [] },
      buildProcessChecker: {
        _log: [Object],
        _checkIntervalMillis: 2000,
        _defaultPattern: /(^|[/\\])(cmake|make|ninja|cl|c\+\+|ld|clang|clang\+\+|gcc|g\+\+|link|icc|armcc|armclang)(-[^/\\]+)?(\.exe)?$/,
        _lastChecked: 1697353308327,
        _finishedP: [Promise],
        _finishedResolver: [Function (anonymous)],
        _timerId: undefined
      }
    },
    name: undefined,
    description: undefined,
    varToValue: [
      { resolve: '${filename}', rule: 'PSubstrings.exe' },
      { resolve: '${relDirpath}', rule: 'build-win32-Debug' },
      {
        resolve: /\$\{filename(?:\[(?:(-?[0-9]+)|(-?[0-9]+)?:(-?[0-9]+)?)\])?\}/,
        rule: [Function: rule]
      },
      {
        resolve: /\$\{relPath(?:\[(?:(-?[0-9]+)|(-?[0-9]+)?:(-?[0-9]+)?)\])?\}/,
        rule: [Function: rule]
      },
      {
        resolve: /\$\{absPath(?:\[(?:(-?[0-9]+)|(-?[0-9]+)?:(-?[0-9]+)?)\])?\}/,
        rule: [Function: rule]
      },
      {
        resolve: /\$\{relDirpath(?:\[(?:(-?[0-9]+)|(-?[0-9]+)?:(-?[0-9]+)?)\])?\}/,
        rule: [Function: rule]
      },
      {
        resolve: /\$\{absDirpath(?:\[(?:(-?[0-9]+)|(-?[0-9]+)?:(-?[0-9]+)?)\])?\}/,
        rule: [Function: rule]
      },
      { resolve: '${extFilename}', rule: '.exe' },
      { resolve: '${baseFilename}', rule: 'PSubstrings' },
      {
        resolve: /\$\{workspaceFolder(?:\[(?:(-?[0-9]+)|(-?[0-9]+)?:(-?[0-9]+)?)\])?\}/,
        rule: [Function: rule]
      },
      {
        resolve: /\$\{workspaceDirectory(?:\[(?:(-?[0-9]+)|(-?[0-9]+)?:(-?[0-9]+)?)\])?\}/,
        rule: [Function: rule]
      },
      { resolve: '${workspaceName}', rule: 'leetCode' },
      { resolve: /\$\{assert(?::([^}]+))?\}/, rule: [Function: rule] },
      { resolve: '${osPathSep}', rule: '\\' },
      { resolve: '${osPathEnvSep}', rule: ';' },
      { resolve: /\$\{config:([^}]+)\}/, rule: [Function: rule] },
      { resolve: /\$\{command:([^}]+)\}/, rule: [AsyncFunction: rule] },
      {
        resolve: /\$\{cwd(?:\[(?:(-?[0-9]+)|(-?[0-9]+)?:(-?[0-9]+)?)\])?\}/,
        rule: [Function: rule]
      }
    ],
    path: 'd:\\Repository\\leetCode\\build-win32-Debug\\PSubstrings.exe',
    options: {
      cwd: 'd:\\Repository\\leetCode\\build-win32-Debug',
      env: {
        ALLUSERSPROFILE: 'C:\\ProgramData',
        APPDATA: 'C:\\Users\\Richard\\AppData\\Roaming',
        APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL: '1',
        CC: 'clang',
        CHROME_CRASHPAD_PIPE_NAME: '\\\\.\\pipe\\crashpad_324_RHQPTQPEUACLXPKD',
        CMAKE_GENERATOR: 'Ninja',
        CMAKE_PREFIX_PATH: ';',
        COMPUTERNAME: 'FX63VD',
        CXX: 'clang++',
        ComSpec: 'C:\\WINDOWS\\system32\\cmd.exe',
        CommonProgramFiles: 'C:\\Program Files\\Common Files',
        'CommonProgramFiles(x86)': 'C:\\Program Files (x86)\\Common Files',
        CommonProgramW6432: 'C:\\Program Files\\Common Files',
        DriverData: 'C:\\Windows\\System32\\Drivers\\DriverData',
        ELECTRON_RUN_AS_NODE: '1',
        GIT_INSTALL_ROOT: 'D:\\scoop\\apps\\git\\current',
        GRADLE_USER_HOME: 'D:\\scoop\\apps\\gradle-bin\\current\\.gradle',
        HOME: 'C:\\Users\\Richard',
        HOMEDRIVE: 'C:',
        HOMEPATH: '\\Users\\Richard',
        JAVA_HOME: 'D:\\scoop\\apps\\oraclejdk-lts\\current',
        JAVA_TOOL_OPTIONS: '-Duser.language=en -Duser.country=US -Duser.variant=US',
        LOCALAPPDATA: 'C:\\Users\\Richard\\AppData\\Local',
        LOGONSERVER: '\\\\FX63VD',
        NUMBER_OF_PROCESSORS: '8',
        NoWriteWithoutFail: '1',
        OPENSSL_ia32cap: ':~0x20000000',
        ORIGINAL_XDG_CURRENT_DESKTOP: 'undefined',
        OS: 'Windows_NT',
        OneDrive: 'F:\\OneDrive',
        OneDriveConsumer: 'F:\\OneDrive',
        PATHEXT: '.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC',
        POSH_THEMES_PATH: 'D:\\scoop\\apps\\oh-my-posh\\current\\themes',
        POWERSHELL_DISTRIBUTION_CHANNEL: 'MSI:Windows 10 Home',
        PROCESSOR_ARCHITECTURE: 'AMD64',
        PROCESSOR_IDENTIFIER: 'Intel64 Family 6 Model 158 Stepping 9, GenuineIntel',
        PROCESSOR_LEVEL: '6',
        PROCESSOR_REVISION: '9e09',
        PSModulePath: 'D:\\scoop\\modules;C:\\Users\\RIchard\\Documents\\WindowsPowerShell\\Modules;',
        PUBLIC: 'C:\\Users\\Public',
        Path: 'C:\\Program Files\\Scoop\\apps\\shims;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files (x86)\\NVIDIA Corporation\\PhysX\\Common;C:\\Program Files\\dotnet\\;D:\\matlab\\bin;C:\\WINDOWS\\System32;C:\\Program Files\\PowerShell\\7\\;D:\\scoop\\apps\\nodejs-lts\\current\\bin;D:\\scoop\\apps\\nodejs-lts\\current;D:\\scoop\\apps\\python\\current\\Scripts;D:\\scoop\\apps\\python\\current;C:\\Program Files\\Scoop\\apps\\shims;D:\\scoop\\shims;D:/scoop\\shims;D:\\msys2\\clang64\\bin;D:\\msys2\\usr\\bin;D:\\scoop\\apps\\oraclejdk-lts\\current\\bin;D:\\scoop\\apps\\vulkan\\current\\Bin;D:\\scoop\\apps\\vulkan\\current\\Tools;D:\\scoop\\apps\\coq\\current;D:\\texlive\\2023\\bin\\windows;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files (x86)\\NVIDIA Corporation\\PhysX\\Common;C:\\Program Files\\dotnet\\;D:\\matlab\\bin;C:\\Users\\Richard\\Default;C:\\Users\\Richard\\.cargo\\bin;C:\\Users\\Richard\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Users\\Richard\\AppData\\Local\\GitHubDesktop\\bin;D:\\npm;D:\\Microsoft VS Code\\bin;D:\\flutter\\bin;D:\\AndroidSDK\\platform-tools;D:\\quickjs\\bin;D:\\NVIDIA GPU Computing Toolkit\\CUDA\\v11.4\\bin;D:\\NVIDIA GPU Computing Toolkit\\CUDA\\v11.4\\extras\\CUPTI\\lib64;D:\\NVIDIA GPU Computing Toolkit\\CUDA\\v11.4\\include;C:\\Users\\Richard\\.dotnet\\tools',
        ProgramData: 'C:\\ProgramData',
        ProgramFiles: 'C:\\Program Files',
        'ProgramFiles(x86)': 'C:\\Program Files (x86)',
        ProgramW6432: 'C:\\Program Files',
        SCOOP: 'D:/scoop',
        SCOOP_GLOBAL: 'C:\\Program Files\\Scoop\\apps',
        SESSIONNAME: 'Console',
        SystemDrive: 'C:',
        SystemRoot: 'C:\\WINDOWS',
        TEMP: 'C:\\Users\\Richard\\AppData\\Local\\Temp',
        TMP: 'C:\\Users\\Richard\\AppData\\Local\\Temp',
        USERDOMAIN: 'FX63VD',
        USERDOMAIN_ROAMINGPROFILE: 'FX63VD',
        USERNAME: 'Richard',
        USERPROFILE: 'C:\\Users\\Richard',
        VK_SDK_PATH: 'D:\\scoop\\apps\\vulkan\\current',
        VSCODE_AMD_ENTRYPOINT: 'vs/workbench/api/node/extensionHostProcess',
        VSCODE_CODE_CACHE_PATH: 'C:\\Users\\Richard\\AppData\\Roaming\\Code\\CachedData\\f1b07bd25dfad64b0167beb15359ae573aecd2cc',
        VSCODE_CRASH_REPORTER_PROCESS_TYPE: 'extensionHost',
        VSCODE_CWD: 'D:\\Microsoft VS Code',
        VSCODE_HANDLES_UNCAUGHT_ERRORS: 'true',
        VSCODE_IPC_HOOK: '\\\\.\\pipe\\eee39ef83ecb98080a934da299e7f94e-1.83.1-main-sock',
        VSCODE_NLS_CONFIG: '{"locale":"zh-cn","osLocale":"zh-cn","availableLanguages":{"*":"zh-cn"},"_languagePackId":"9f233b0c8bdef919c90298683f3dac6b.zh-cn","_translationsConfigFile":"C:\\\\Users\\\\Richard\\\\AppData\\\\Roaming\\\\Code\\\\clp\\\\9f233b0c8bdef919c90298683f3dac6b.zh-cn\\\\tcf.json","_cacheRoot":"C:\\\\Users\\\\Richard\\\\AppData\\\\Roaming\\\\Code\\\\clp\\\\9f233b0c8bdef919c90298683f3dac6b.zh-cn","_resolvedLanguagePackCoreLocation":"C:\\\\Users\\\\Richard\\\\AppData\\\\Roaming\\\\Code\\\\clp\\\\9f233b0c8bdef919c90298683f3dac6b.zh-cn\\\\f1b07bd25dfad64b0167beb15359ae573aecd2cc","_corruptedFile":"C:\\\\Users\\\\Richard\\\\AppData\\\\Roaming\\\\Code\\\\clp\\\\9f233b0c8bdef919c90298683f3dac6b.zh-cn\\\\corrupted.info","_languagePackSupport":true}',
        VSCODE_PID: '324',
        VULKAN_SDK: 'D:\\scoop\\apps\\vulkan\\current',
        WindowsSDKDir: 'D:\\Microsoft Visual Studio\\Windows Kits\\10',
        ZES_ENABLE_SYSMAN: '1',
        windir: 'C:\\WINDOWS'
      }
    },
    _frameworkSpecific: {},
    markAsSkipped: false,
    runTask: { before: [], beforeEach: [], after: [], afterEach: [] },
    spawner: s {},
    resolvedSourceFileMap: {},
    log: {
      _logger: {
        configSection: 'testMate.cpp.log',
        workspaceFolder: undefined,
        outputChannelName: 'C++ TestMate',
        inspectOptions: [Object],
        includeLocation: false,
        targets: [Array],
        nextInspectOptions: undefined,
        configChangeSubscription: [Object]
      }
    },
    workspaceFolder: {
      uri: s {
        scheme: 'file',
        authority: '',
        path: '/d:/Repository/leetCode',
        query: '',
        fragment: '',
        _formatted: 'file:///d%3A/Repository/leetCode',
        _fsPath: 'd:\\Repository\\leetCode'
      },
      name: 'leetCode',
      index: 0
    },
    workspacePath: 'd:\\Repository\\leetCode',
    testController: {
      controller: {
        items: [Object],
        label: [Getter/Setter],
        refreshHandler: [Getter/Setter],
        id: [Getter],
        createRunProfile: [Function: createRunProfile],
        createTestItem: [Function: createTestItem],
        createTestRun: [Function: createTestRun],
        invalidateTestResults: [Function: invalidateTestResults],
        resolveHandler: [Getter/Setter],
        dispose: [Function: dispose]
      },
      testItem2test: WeakMap { <items unknown> }
    },
    cancellationToken: R { a: false, b: null },
    taskPool: { _maxTaskCount: 1, _runningTaskCount: 1, _waitingTasks: [] },
    executeTask: [Function (anonymous)],
    parallelizationPool: { _maxTaskCount: 1, _runningTaskCount: 1, _waitingTasks: [] },
    optionsHash: '130044'
  },
  frameworkName: 'GoogleTest',
  frameworkVersion: undefined,
  log: {
    _logger: {
      configSection: 'testMate.cpp.log',
      workspaceFolder: undefined,
      outputChannelName: 'C++ TestMate',
      inspectOptions: { depth: 3 },
      includeLocation: false,
      targets: [ [c] ],
      nextInspectOptions: undefined,
      configChangeSubscription: { dispose: [Function (anonymous)] }
    }
  },
  _lastReloadTime: 1697353302803.611,
  _tests: Map(5) {
    'naive.0' => s {
      exec: [Circular *1],
      id: 'naive.0',
      _skipped: false,
      _staticError: undefined,
      _tags: [],
      _frameworkTag: qi { id: 'framework.gtest' },
      debuggable: true,
      runnable: true,
      subLevel: 0,
      log: { _logger: [Object] },
      _skipReported: false,
      _subTests: undefined,
      _item: s {
        id: 'naive.0',
        uri: [s],
        children: [Object],
        range: [Getter/Setter],
        label: [Getter/Setter],
        description: [Getter/Setter],
        sortText: [Getter/Setter],
        canResolveChildren: [Getter/Setter],
        busy: [Getter/Setter],
        error: [Getter/Setter],
        tags: [Getter/Setter]
      }
    },
    'naive.1' => s {
      exec: [Circular *1],
      id: 'naive.1',
      _skipped: false,
      _staticError: undefined,
      _tags: [],
      _frameworkTag: qi { id: 'framework.gtest' },
      debuggable: true,
      runnable: true,
      subLevel: 0,
      log: { _logger: [Object] },
      _skipReported: false,
      _subTests: undefined,
      _item: s {
        id: 'naive.1',
        uri: [s],
        children: [Object],
        range: [Getter/Setter],
        label: [Getter/Setter],
        description: [Getter/Setter],
        sortText: [Getter/Setter],
        canResolveChildren: [Getter/Setter],
        busy: [Getter/Setter],
        error: [Getter/Setter],
        tags: [Getter/Setter]
      }
    },
    'naive.2' => s {
      exec: [Circular *1],
      id: 'naive.2',
      _skipped: false,
      _staticError: undefined,
      _tags: [],
      _frameworkTag: qi { id: 'framework.gtest' },
      debuggable: true,
      runnable: true,
      subLevel: 0,
      log: { _logger: [Object] },
      _skipReported: false,
      _subTests: undefined,
      _item: s {
        id: 'naive.2',
        uri: [s],
        children: [Object],
        range: [Getter/Setter],
        label: [Getter/Setter],
        description: [Getter/Setter],
        sortText: [Getter/Setter],
        canResolveChildren: [Getter/Setter],
        busy: [Getter/Setter],
        error: [Getter/Setter],
        tags: [Getter/Setter]
      }
    },
    'naive.3' => s {
      exec: [Circular *1],
      id: 'naive.3',
      _skipped: false,
      _staticError: undefined,
      _tags: [],
      _frameworkTag: qi { id: 'framework.gtest' },
      debuggable: true,
      runnable: true,
      subLevel: 0,
      log: { _logger: [Object] },
      _skipReported: false,
      _subTests: undefined,
      _item: s {
        id: 'naive.3',
        uri: [s],
        children: [Object],
        range: [Getter/Setter],
        label: [Getter/Setter],
        description: [Getter/Setter],
        sortText: [Getter/Setter],
        canResolveChildren: [Getter/Setter],
        busy: [Getter/Setter],
        error: [Getter/Setter],
        tags: [Getter/Setter]
      }
    },
    'naive.4' => s {
      exec: [Circular *1],
      id: 'naive.4',
      _skipped: false,
      _staticError: undefined,
      _tags: [],
      _frameworkTag: qi { id: 'framework.gtest' },
      debuggable: true,
      runnable: true,
      subLevel: 0,
      log: { _logger: [Object] },
      _skipReported: false,
      _subTests: undefined,
      _item: s {
        id: 'naive.4',
        uri: [s],
        children: [Object],
        range: [Getter/Setter],
        label: [Getter/Setter],
        description: [Getter/Setter],
        sortText: [Getter/Setter],
        canResolveChildren: [Getter/Setter],
        busy: [Getter/Setter],
        error: [Getter/Setter],
        tags: [Getter/Setter]
      }
    }
  },
  _execItem: m {
    executable: [Circular *1],
    _busyCounter: 0,
    _item: s {
      id: 'd:\\Repository\\leetCode\\build-win32-Debug\\PSubstrings.exe#130044',
      uri: undefined,
      children: {
        size: [Getter],
        forEach: [Function: forEach],
        replace: [Function: replace],
        add: [Function: add],
        delete: [Function: delete],
        get: [Function: get],
        toJSON: [Function: toJSON],
        [Symbol(Symbol.iterator)]: [Function: [Symbol.iterator]]
      },
      range: [Getter/Setter],
      label: [Getter/Setter],
      description: [Getter/Setter],
      sortText: [Getter/Setter],
      canResolveChildren: [Getter/Setter],
      busy: [Getter/Setter],
      error: [Getter/Setter],
      tags: [Getter/Setter]
    },
    _itemForStaticError: undefined,
    _lock: Promise { undefined }
  },
  _argumentPrefix: 'gtest_',
  _createAndAddTest: [AsyncFunction (anonymous)]
} a {
  test: <ref *1> s {
    exec: <ref *2> f {
      shared: {
        shared: [Object],
        name: undefined,
        description: undefined,
        varToValue: [Array],
        path: 'd:\\Repository\\leetCode\\build-win32-Debug\\PSubstrings.exe',
        options: [Object],
        _frameworkSpecific: {},
        markAsSkipped: false,
        runTask: [Object],
        spawner: s {},
        resolvedSourceFileMap: {},
        log: [Object],
        workspaceFolder: [Object],
        workspacePath: 'd:\\Repository\\leetCode',
        testController: [Object],
        cancellationToken: [R],
        taskPool: [Object],
        executeTask: [Function (anonymous)],
        parallelizationPool: [Object],
        optionsHash: '130044'
      },
      frameworkName: 'GoogleTest',
      frameworkVersion: undefined,
      log: { _logger: [Object] },
      _lastReloadTime: 1697353302803.611,
      _tests: Map(5) {
        'naive.0' => [Circular *1],
        'naive.1' => [s],
        'naive.2' => [s],
        'naive.3' => [s],
        'naive.4' => [s]
      },
      _execItem: m {
        executable: [Circular *2],
        _busyCounter: 0,
        _item: [s],
        _itemForStaticError: undefined,
        _lock: [Promise]
      },
      _argumentPrefix: 'gtest_',
      _createAndAddTest: [AsyncFunction (anonymous)]
    },
    id: 'naive.0',
    _skipped: false,
    _staticError: undefined,
    _tags: [],
    _frameworkTag: qi { id: 'framework.gtest' },
    debuggable: true,
    runnable: true,
    subLevel: 0,
    log: {
      _logger: {
        configSection: 'testMate.cpp.log',
        workspaceFolder: undefined,
        outputChannelName: 'C++ TestMate',
        inspectOptions: [Object],
        includeLocation: false,
        targets: [Array],
        nextInspectOptions: undefined,
        configChangeSubscription: [Object]
      }
    },
    _skipReported: false,
    _subTests: undefined,
    _item: s {
      id: 'naive.0',
      uri: s {
        scheme: 'file',
        authority: '',
        path: '/D:/Repository/leetCode/src/PSubstrings/naive.hpp',
        query: '',
        fragment: '',
        _formatted: null,
        _fsPath: null
      },
      children: {
        size: [Getter],
        forEach: [Function: forEach],
        replace: [Function: replace],
        add: [Function: add],
        delete: [Function: delete],
        get: [Function: get],
        toJSON: [Function: toJSON],
        [Symbol(Symbol.iterator)]: [Function: [Symbol.iterator]]
      },
      range: [Getter/Setter],
      label: [Getter/Setter],
      description: [Getter/Setter],
      sortText: [Getter/Setter],
      canResolveChildren: [Getter/Setter],
      busy: [Getter/Setter],
      error: [Getter/Setter],
      tags: [Getter/Setter]
    }
  },
  testRun: {
    isPersisted: true,
    token: R { a: true, b: null },
    name: undefined,
    coverageProvider: [Getter/Setter],
    enqueued: [Function (anonymous)],
    skipped: [Function (anonymous)],
    started: [Function (anonymous)],
    errored: [Function (anonymous)],
    failed: [Function (anonymous)],
    passed: [Function (anonymous)],
    appendOutput: [Function: appendOutput],
    end: [Function: end]
  },
  runPrefix: '\x1B[2m$1│ \x1B[22m',
  addBeginEndMsg: false,
  level: 0,
  log: {
    _logger: {
      configSection: 'testMate.cpp.log',
      workspaceFolder: undefined,
      outputChannelName: 'C++ TestMate',
      inspectOptions: { depth: 3 },
      includeLocation: false,
      targets: [ [c] ],
      nextInspectOptions: undefined,
      configChangeSubscription: { dispose: [Function (anonymous)] }
    }
  },
  _messages: [],
  _result: undefined,
  _duration: undefined,
  _built: false,
  _subTestResultBuilders: []
};
[2023-10-15 07:02:03.570] [INFO] Test naive.0 has stopped.
[2023-10-15 07:02:03.570] [INFO] proc finished: d:\Repository\leetCode\build-win32-Debug\PSubstrings.exe
[2023-10-15 07:02:03.570] [INFO] reloadTests GoogleTest undefined d:\Repository\leetCode\build-win32-Debug\PSubstrings.exe
[2023-10-15 07:02:03.570] [DEBUG] reloadTests was skipped due to mtime d:\Repository\leetCode\build-win32-Debug\PSubstrings.exe
```

</details>

**Comment by matepek** [comment]
Hello,
I believe it is vscode related. Please open an issue on their repo.

**Comment by connor4312** [comment]
Hi, following up from VS Code.

A while back UI behavior in VS Code changed a bit. Clicking the "stop" button cancels the `TestRun.token`. However, the UI will continue showing the running state until the extension calls `TestRun.end()`.

There's no known issues relating to cancellation in VS Code itself, so I believe that's probably what's happening here.

**Comment by matepek** [comment]
`onCancellationRequested` is not fired at https://github.com/matepek/vscode-catch2-test-adapter/blob/master/src/main.ts#L177.
I think it is a vscode issue @connor4312

**Comment by connor4312** [comment]
Note that the cancellation token provided to the runHandler is different than the one on the TestRunRequest. This is important because the request for a continuous test run will not be cancelled until the user turns continuous test run off, while individual test runs may still be cancelled using the "stop" button.

This is not the first time someone has been tripped up by this different, I will clarify in our API docs.

**Comment by matepek** [comment]
<details>
<summary>Fixed in <b>v4.6.1</b>.</summary>

This issue was mentioned in [CHANGELOG.md](./CHANGELOG.md) under a released entry so it is assumed to be fixed.
User verifications are always welcome.
</details>
